### PR TITLE
Customer and Customer Repo

### PR DESCRIPTION
**Pull request by ajongaro**
Fully built customer and customer repository classes, with rspec passing on full project up to this point in history.
Spec harness passes on it0-it2

### CODE REVIEW

File: lib/customer_repository.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  def update(id, attributes)
+    sanitized_attributes = {
+      first_name: attributes[:first_name],
+      last_name: attributes[:last_name],
+      updated_at: Time.now
+    }
+    super(id, sanitized_attributes)
+  end
+
+end
```
Comment (ajongaro) [inline lib/customer_repository.rb]: Will address this in next commit.

File: lib/customer_repository.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require_relative 'repository'
+
+class CustomerRepository < Repository
+
```
Comment (ajongaro) [inline lib/customer_repository.rb]: Fixing now.

File: spec/customer_repository_spec.rb [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+  describe 'create(attributes)' do
+    it 'creates a new Customer instance with provided attributes' do
+      expect(cr.all).to eq([]) 
+
+      cr.add_to_repo(customer_1)
+      expect(cr.all.count).to eq(1)
+
+      cr.create({ 
+        first_name: 'Tron',
```
[the remark this answers is not in the document]
Comment (ajongaro) [reply spec/customer_repository_spec.rb:95]: Need clarification/further information on how to style these blocks of hashes.

File: spec/customer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                                    updated_at:  Time.now }) }
+
+  let(:customer_2) { Customer.new({ :id => 2,
+                                    :first_name => 'Michael',
+                                    :last_name => 'Jackson',
+                                    :created_at => Time.now,
+                                    :updated_at => Time.now }) }
+
+  let(:customer_3) { Customer.new({ :id => 8,
+                                    :first_name => 'Argo',
```
Comment (ajongaro) [inline spec/customer_spec.rb]: Fixing this now.

File: spec/customer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                                    :last_name => 'Angus',
```
Comment (ajongaro) [inline spec/customer_spec.rb]: Addressing this now before push.

File: spec/customer_spec.rb [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+                                    first_name: 'Joan',
+                                    last_name: 'Clarke',
+                                    created_at:  Time.now,
+                                    updated_at:  Time.now }) }
+
+  let(:customer_2) { Customer.new({ :id => 2,
+                                    :first_name => 'Michael',
+                                    :last_name => 'Jackson',
+                                    :created_at => Time.now,
+                                    :updated_at => Time.now }) }
```
Comment (ajongaro) [inline spec/customer_spec.rb]: Fixing now.

File: spec/customer_repository_spec.rb [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        created_at:  Time.now,
+        updated_at:  Time.now })
+
+      expect(cr.all.count).to eq(3)
+      expect(cr.all[2].first_name).to eq('Kathy')
+      expect(cr.all[2].last_name).to eq('Peterson')
+    end
+  end
+
+  describe 'update(attributes)' do
```
Comment (this-is-joeking) [inline spec/customer_repository_spec.rb:116]: Seems like we should also include a test for update that checks that updated at time has changed pon

**Review by this-is-joeking — CHANGES_REQUESTED** [review]
Only change requested is to add test for updated at time changing

**Review by this-is-joeking — APPROVED** [review]
Looks great

**Pull request merged**